### PR TITLE
Add Coq.8.20.0

### PR DESCRIPTION
**Pull request by proux01**
Cc @silene (coRM) and @palmskog (who did many previous Coq packages)

### CODE REVIEW

File: packages/coq-core/coq-core.8.20.0/opam [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+-noinit options, as will otherwise fail to find the regular Coq
+prelude, now living in the coq-stdlib package."""
+maintainer: ["The Coq development team <coqdev@inria.fr>"]
+authors: ["The Coq development team, INRIA, CNRS, and contributors"]
+license: "LGPL-2.1-only"
+homepage: "https://coq.inria.fr/"
+doc: "https://coq.github.io/doc/"
+bug-reports: "https://github.com/coq/coq/issues"
+depends: [
+  "dune" {>= "3.6" & >= "3.6.1"}
```
Comment (palmskog) [inline packages/coq-core/coq-core.8.20.0/opam:28]: @proux01 the first constraint looks redundant to me, shouldn't it be just `>= "3.6.1"`?

Comment (proux01) [reply packages/coq-core/coq-core.8.20.0/opam:28]: Copied it from the OPAM file in Coq repo, apparently the result of some dune issue https://github.com/coq/coq/blob/master/dune-project#L24-L28 but we could indeed simplify it.

**Comment by proux01** [comment]
@MSoegtropIMC you're certainly our best Windows expert, does the CI error here ring a bell:
```
Uncaught exception: Invalid_argument("index out of bounds")
Error while running 'D:\opamroot\default\bin/ocamlfind.exe ocamlc -version' (exit code 3)
Configuration script failed!
```

**Comment by MSoegtropIMC** [comment]
No, sorry. If the same stuff works elsewhere it is likely an issue with handling absolute paths. Note that in Coq Platform I am using a patched version of ocamlfind which handles paths very differently. The main reason for the patch is to make ocamlfind relocatable, but it is also avoids a lot of issues with paths on Windows.

**Comment by proux01** [comment]
Ok, thanks for the answer.
By the way, do you have any plan to sbmit you ocamlfind patch upstream?

**Comment by MSoegtropIMC** [comment]
I proposed the patch in https://github.com/ocaml/ocamlfind/issues/60 but OCaml wants to follow a different approach (which does not work for Coq Platform - at least not without quite a bit of extra work).

**Comment by MSoegtropIMC** [comment]
I reviewed the discussion there and I still think we won't converge on this. Making Windows / MacOS installers for OCaml software still seems to be a niche application.

**Comment by MSoegtropIMC** [comment]
@proux01 : any idea why it did work for 8.19? Also note that Coq Platform does build Coq via opam on Windows. I think we should at least test if this works with the patched ocamlfind.

**Comment by proux01** [comment]
Are you sure it worked for 8.19? My understanding is that OPAM for windows is a recent thing and it just wan't tested then.

**Comment by MSoegtropIMC** [comment]
Coq Platform builds Coq using opam on Windows since 8.13, so yes I am sure this works. But we are using cygwin as build host - otherwise many opam packages won't build because they need a Posix shell (e.g. for configure scripts).

**Comment by MSoegtropIMC** [comment]
See e.g. https://github.com/coq/platform/actions/runs/10693583167/job/29643963688.

**Comment by proux01** [comment]
Yes, that's what I meant, Coq CI for windows based on the platform is green but this precise OPAM CI was probably just never tested before.

**Comment by MSoegtropIMC** [comment]
Indeed if this is a direct opam on WIndows CI I wouldn't expect that it works. How far does it get?

**Comment by proux01** [comment]
As you can see, it doesn't go past `ocamlc -version` in the configure script.

**Comment by avsm** [comment]
This is indeed a new test, and not a blocker. The other builds look fine to me to merge.

**Comment by mseri** [comment]
Thanks

**Comment by proux01** [comment]
Thanks!

**Pull request merged**